core/vtk/ttkAlgorithm/ttkAlgorithm.cpp

```diff
@@ -683,32 +708,41 @@ void ttkAlgorithm::MPIPipelinePreconditioning(
 
   bool pointValidity{false};
   bool cellValidity{false};
-  if(globalPointIds != nullptr) {
-    unsigned char *ghostPoints = ttkUtils::GetPointer<unsigned char>(
-      input->GetPointData()->GetArray("vtkGhostType"));
-    pointValidity
-      = checkGlobalIdValidity(globalPointIds, vertexNumber, ghostPoints);
+  if((triangulation != nullptr
+      && triangulation->getType() == ttk::Triangulation::Type::EXPLICIT)
```

Comment:
thanks for also considering the case `ttk::Triangulation::Type::Compact` everywhere there is a special treatment for the explicit version (the compact version is a variant of the explicit one).